pkg/cmd/pulumi/preview.go

```diff
@@ -92,7 +146,7 @@ func buildImportFile(events <-chan engine.Event) *promise.Promise[importFile] {
 			var parent string
 			if new.Parent != "" {
 				// If the parent is just the root stack then skip it as we don't need to import that.
-				if new.Parent.Type() != resource.RootStackType {
+				if new.Parent.QualifiedType() != resource.RootStackType {
```

Comment:
Substacks means this should check qualified type not just type. Missed that when adding this.